api/base/filters.py

```diff
@@ -20,53 +30,184 @@ def filter_queryset(self, request, queryset, view):
+            }]
+        else:  # TODO: let times be as generic as possible (i.e. whole month, whole year)
+            start = self.convert_value(value, field)
+            stop = start + datetime.timedelta(days=1)
+            return [{
+                'op': 'gte',
+                'value': start
+            }, {
+                'op': 'lt',
+                'value': stop
+            }]
+
+    def parse_query_params(self, query_params):
+        """Maps query params to a dict useable for filtering
+        :param dict query_params:
+        :return dict: of the format {
+            <resolved_field_name>: {
+                'op': <comparison_operator>,
+                'value': <resolved_value>
+            }
+        }
+        """
+        query = {}
+        for key, value in query_params.iteritems():
+            match = self.QUERY_PATTERN.match(key)
+            if match:
+                field_name = match.groupdict()['field'].strip()
+                field = self._get_field_or_error(field_name)
+
+                op = match.groupdict().get('op') or self._get_default_operator(field)
+                self._validate_operator(field, field_name, op)
+
+                field_name = self.convert_key(field_name, field)
+                if field_name not in query:
+                    query[field_name] = []
+
+                # Special case date(time)s to allow for ambiguous date matches
+                if type(field) in self.DATE_FIELDS:
+                    query[field_name].extend(self._parse_date_param(field, field_name, op, value))
+                else:
+                    query[field_name].append({
+                        'op': op,
+                        'value': self.convert_value(value, field)
+                    })
+        return query
 
-    # Used so that that queries by _id will work
-    def convert_key(self, key):
-        key = key.strip()
-        if self.serializer_class._declared_fields[key].source:
-            return self.serializer_class._declared_fields[key].source
-        return key
+    def convert_key(self, field_name, field):
+        """Used so that that queries on fields with the souce attribute set will work
+        :param basestring field_name: text representation of the field name
+        :param rest_framework.fields.Field field: Field instance
+        """
+        return field.source or field_name
 
-    # Used to convert string values from query params to Python booleans when necessary
     def convert_value(self, value, field):
-        field_type = type(self.serializer_class._declared_fields[field])
-        value = value.strip()
-        if field_type == ser.BooleanField:
+        """Used to convert string values from query params to bools and dates when necessary
+        :param basestring value: value to be resolved
+        :param rest_framework.fields.Field field: Field instance
+        """
+        if isinstance(field, ser.BooleanField):
```

Comment:
> See notes on isinstance above for calls **in this method**
(emphasis added)

That line just happened to be at the top of the method, and I made one comment at the top instead of making multiple comments on separate lines. I, er... thought it would save time and energy.

See Lines 194 and Line 202, which are both nested under `convert_value`